src/commands/logout.js

```diff
@@ -1,16 +1,43 @@
-import auth from '../utils/auth';
+/**
+ * Default logout URL if you use nextjs-auth0's `handleAuth` in
+ * pages/api/[...auth0].ts
+ *
+ * Docs
+ * @see https://auth0.github.io/nextjs-auth0/modules/handlers_auth.html
+ * @see https://auth0.github.io/nextjs-auth0/modules/handlers_logout.html
+ */
+const DEFAULT_LOGOUT_URL = '/api/auth/logout';
 
-Cypress.Commands.add('logout', (returnTo) => {
-  const options = {
-    clientID: Cypress.env('auth0ClientId'),
-  };
+Cypress.Commands.add('logout', (options = {}) => {
+  const { logoutUrl, returnTo } = options;
 
-  if (returnTo) {
-    options.returnTo = returnTo;
+  /**
+   * nextjs-auth0's logout URL supports redirecting to a custom `returnTo` URL.
+   * cy.logout supports that and will create a logout URL that either uses the
+   * current hostname or the `returnTo` URLs one.
+   *
+   * @see https://auth0.github.io/nextjs-auth0/interfaces/handlers_logout.logoutoptions.html
+   */
+  const preparedLogoutUrl = new URL(
+    logoutUrl || DEFAULT_LOGOUT_URL,
+    window.location.href,
+  );
+
+  if (typeof returnTo === 'string') {
+    preparedLogoutUrl.searchParams.set('returnTo', returnTo);
   }
 
-  const logoutUrl = auth.client.buildLogoutUrl(options);
```

Comment:
Given the confusion with `logutUrl` (sorry!!), I think this is the best approach in order to simplify the API (i.e. prevent unnecessary breaking changes) and maximize compatibility with the auth client:

```js
/**
 * Default logout URL if you use nextjs-auth0's `handleAuth` in
 * pages/api/[...auth0].ts
 *
 * Docs
 * @see https://auth0.github.io/nextjs-auth0/modules/handlers_auth.html
 * @see https://auth0.github.io/nextjs-auth0/modules/handlers_logout.html
 */
const DEFAULT_LOGOUT_URL = '/api/auth/logout';

Cypress.Commands.add('logout', (returnTo) => {
  const options = {
    clientID: Cypress.env('auth0ClientId'),
  };

  if (returnTo) {
    options.returnTo = returnTo;
  }

  const logoutUrl = auth.client.buildLogoutUrl(options);

  cy.request(logoutUrl);
  cy.clearAuth0Cookies();
});
```

I haven't tested this code so treat it as an idea/example.

What do you think of this approach?

Reply:
Or if we were to use your new URL-building method then the example would look like this:

```js
/**
 * Default logout URL if you use nextjs-auth0's `handleAuth` in
 * pages/api/[...auth0].ts
 *
 * Docs
 * @see https://auth0.github.io/nextjs-auth0/modules/handlers_auth.html
 * @see https://auth0.github.io/nextjs-auth0/modules/handlers_logout.html
 */
const DEFAULT_LOGOUT_URL = '/api/auth/logout';

Cypress.Commands.add('logout', (returnTo) => {
 const logoutUrl = new URL(
    DEFAULT_LOGOUT_URL,
    window.location.href,
  );

  if (typeof returnTo === 'string') {
    logoutUrl.searchParams.set('returnTo', returnTo);
  }

  cy.request(logoutUrl);
  cy.clearAuth0Cookies();
});
```

I guess my may point is that you may think it's more important to prevent a breaking change instead of supporting the`logoutUrl` param

Reply:
Thanks for your input and discussion on Discord. I simplified it and removed the breaking change. It is now compatible with the current version and allows customizing it for all cases though: https://github.com/sir-dunxalot/cypress-nextjs-auth0/pull/44/commits/44b08d40b692222fe13916304da41f989ccb6cae.